neuro_san/coded_tools/math_guy/async_closeable.py

```diff
@@ -0,0 +1,27 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from logging import getLogger
+from logging import Logger
+
+
+class AsyncCloseable:
+    """
+    A simple object used to test the asynchronous close()-ing of objects on sly_data.
+    """
```

Comment:
Used for testing asynchronous close()-ing of sly_data

Reply:
There is a fine line here that I don't yet know what to do about.
Some of the networks here are bare-bones examples and some are tests.

Maybe the only networks/coded_tools in neuro-san should be the test infrastructure ones
and the rest moved to the test/directory. That would require:
* Some discussion as to which examples need to be promoted to neuro-san-studio (maybe none)
* Some infrastructure so that networks/coded tool paths could come from > 1 place, more like a PYTHONPATH.

That's out of scope here.